.github/workflows/tests.yaml

```diff
@@ -0,0 +1,114 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: tests
+on:
+  pull_request:
+  pull_request_target:
+    types: [labeled]
+jobs:
+  unit:
+    if: "${{ github.event.action != 'labeled' || github.event.label.name == 'tests: run' }}"
+    name: unit tests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Remove PR Label
+        if: "${{ github.event.action == 'labeled' && github.event.label.name == 'tests: run' }}"
+        uses: actions/github-script@v6
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            try {
+              await github.rest.issues.removeLabel({
+                name: 'tests: run',
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                issue_number: context.payload.pull_request.number
+              });
+            } catch (e) {
+              console.log('Failed to remove label. Another job may have already removed it!');
+            }
+      - name: Setup Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: '1.20'
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
+      - name: Set up build.env with phony secrets.
+        run: cp build.sample.env build.env
+      - name: Verify no changes from code generation. If this check has failed, run `make generate` and commit the changes.
+        run: "make generate \nif ! git diff --exit-code --quiet ; then \n  echo\n  echo\n  git diff --stat \"HEAD\"\n  echo\n  echo\n  echo 'ERROR: Lint tools caused changes to the working dir. '\n  exit 1\nfi\n"
+      - name: make test
+        run: make test
+  e2e:
+    needs: unit
+    if: "${{ github.event.action == 'labeled' && github.event.label.name == 'tests: run' }}"
```

Comment:
This now means: "only run e2e when the label `tests:run` is added"

Reply:
this will cause the `unit tests` to run twice i believe on internal PRs. Once when you open the PR and once when you add the label. Not a big deal I don't think though

Reply:
That's ok with me.